src/components/ListItem.jsx

```diff
@@ -1,23 +1,39 @@
 import './ListItem.css';
 import { updateItem } from '../api';
 import { sub } from 'date-fns';
+import { getFutureDate } from '../utils';
+import { getDaysBetweenDates } from '../utils/dates';
+import { calculateEstimate } from '@the-collab-lab/shopping-list-utils';
 
 export function ListItem({ item, listToken }) {
 	const handleUpdate = async (isChecked) => {
 		let day = Date.now();
 		let purchaseCounter = item.totalPurchases + 1;
-
 		try {
 			if (isChecked) {
-				const newData = { day, purchaseCounter };
+				const lastEstimate = Number.isNaN(item.dateNextPurchased)
+					? undefined
+					: item.dateNextPurchased;
```

Comment:
Since the `lastEstimate` should return the previously recorded time frame in which the item is to be bought i.e difference between `dateNextPurchased` and `dateLastPurchased`, I think the value for `lastEstimate` here is not correct. I think the right calculation should be `const lastEstimate = getDaysBetweenDays(item.dateLastPurchased, item.dateNextPurchased`

To do the above calculation, you may need to convert the value of `item.dateLastPurchased` to an actual date using the `new Date()` function

I think that is the reason I am getting this error below in the console
![image](https://user-images.githubusercontent.com/79487718/236671976-b0444484-db8e-4f76-921a-d73559511b13.png)

When I try to check this value
![image](https://user-images.githubusercontent.com/79487718/236672011-8380b077-3c4c-4b7b-b6a7-af163da41cae.png)

I don't get this error when I create a new item and check it

Reply:
Well done Sandra and Cassandra - this issue is blowing my mind just thinking about it 🤯 

I am afraid I get the same as Mike - ticking an existing item throws an error.

Reply:
What mike suggests is correct. If the `dateNextPurchased` is not NaN you need to get the previous estimation in dates which is the difference between the `dateLastPurchased` and the `dateNextPurchased`.